Tensile/KernelWriter.py

```diff
@@ -140,33 +143,19 @@ def makeSchedule(self, kernel, tensorParametersA, tensorParametersB, localWriteE
       if schedDb & 0x1:
         print("makeSchedule-gr, readCnt=", readCnt, "firstStep=", firstStep, "endIter=", endIter)
 
-      #front load dtlsM0UpdateACode code for A loads
-      self.perIterGlobalReadCode[0].addCode(self.dtlsM0UpdateACode)
+      # insert dtlsM0UpdateACode dtlsM0UpdateBCode code
+      if self.globalReadACode.middle.items():
+        self.globalReadACode.middle.items()[0].items().insert(0,self.dtlsM0UpdateACode)
+      if self.globalReadBCode.middle.items():
+        self.globalReadBCode.middle.items()[0].items().insert(0,self.dtlsM0UpdateBCode)
       # append 'n' global load at a time 
       # append global load(S) first 'number of global load(s) determined by  firstStep
-      # insert dtlsM0UpdateBCode for B loads if firstStep > #A loads
-      if firstStep > self.globalReadACode.middle.countType(Code.GlobalReadInst):
-         readACnt = self.globalReadACode.middle.countType(Code.GlobalReadInst)
-         for item in itemsToSched[:readACnt]:
-           self.perIterGlobalReadCode[0].addCode(item)
-         self.perIterGlobalReadCode[0].addCode(self.dtlsM0UpdateBCode) 
-         schedDtls = 1
-         itemsToSched = itemsToSched[readACnt:]
-         readACnt = firstStep - readACnt
-         for item in itemsToSched[:readACnt]:
-           self.perIterGlobalReadCode[0].addCode(item)
-         itemsToSched = itemsToSched[readACnt:]
-      else:
-         for item in itemsToSched[:firstStep]:
-           self.perIterGlobalReadCode[0].addCode(item)
-         itemsToSched = itemsToSched[firstStep:]
+      for item in itemsToSched[:firstStep]:
+        self.perIterGlobalReadCode[0].addCode(item)
+      itemsToSched = itemsToSched[firstStep:]
```

Comment:
@ramjana 
above change is related dtl, original code is trying to find first globalReadACode and first globalReadBCode, then insert M0update to perIterGlobalReadCode before them. 
because schedule algorithm is to schedule globalRead instruction in perIterGlobalReadCode with mfma
however, insert M0update to perIterGlobalReadCode will also schedule M0update with mfma.
I try to avoid schedule M0update by inserting it directly into first globalReadACode and first globalReadBCode.

origin:
M0update | mfma | GRA | mfma | M0update | mfma | GRB

modified:
M0update+GRA | mfma | M0update+GRB | mfma | mfma